lib/ApiSession.ts

```diff
@@ -209,84 +204,69 @@ export class ApiSession implements SolidityAddressable {
   }
 
   /**
-   * Queries/Executes a contract function, capable of returning the {@link ContractFunctionResult} if successful. This depends on the {@param returnType} of course.
+   * Queries/Executes a contract function, capable of returning the {@link StratoContractCallResponse} if successful.
    */
-  public async execute<T extends TypeOfExecutionReturn>(
-    transaction: ContractFunctionCall,
-    returnType?: T,
-    getReceipt?: boolean
-  ): Promise<ExecutionReturnTypes<ContractFunctionResult>[T]>;
+  public async execute(
+    transaction: ContractExecuteTransaction | ContractCreateTransaction
+  ): Promise<StratoContractCallResponse>;
 
   /**
-   * A catch-all/generic {@link Transaction} execution operation yielding, upon success, of a {@link TransactionResponse}.
+   * A catch-all/generic {@link Transaction} execution operation yielding, upon success, of a {@link StratoTransactionResponse}.
    */
-  public async execute<T extends TypeOfExecutionReturn>(
-    transaction: Transaction,
-    returnType?: T,
-    getReceipt?: boolean
-  ): Promise<ExecutionReturnTypes<TransactionResponse>[T]>;
+  public async execute(
+    transaction: Transaction
+  ): Promise<StratoTransactionResponse>;
 
   /**
    * A catch-all/generic {@link Query<R>} execution operation yielding, upon success, of the underlying generic-bounded response type, R.
    */
-  public async execute<T extends TypeOfExecutionReturn, R>(
-    transaction: Query<R>,
-    returnType?: T,
-    getReceipt?: boolean
-  ): Promise<ExecutionReturnTypes<R>[T]>;
+  public async execute<R>(transaction: Query<R>): Promise<R>;
 
   // Overload implementation
-  public async execute<T extends TypeOfExecutionReturn, R>(
-    transaction: SessionExecutable<R>,
-    returnType: T,
-    getReceipt = false
-  ): Promise<ExecutionReturnTypes<AllowedExecutionReturnTypes<R>>[T]> {
-    const isContractTransaction =
-      transaction instanceof ContractCallQuery ||
+  public async execute<R>(
```

Comment:
Very happy that we managed to simplify so much around the app with this mirror integration and at the same time provide more value for less.